core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -421,6 +420,51 @@ private[spark] class AppStatusStore(
     constructTaskDataList(taskDataWrapperIter)
   }
 
+  def exceptionSummary(stageId: Int, attemptId: Int): Seq[v1.ExceptionSummary] = {
+    val (stageData, _) = stageAttempt(stageId, attemptId)
+    val key = Array(stageId, attemptId, stageData.numFailedTasks)
+    asOption(store.read(classOf[CachedExceptionSummary], key))
```

Comment:
If the exception summary is cached when the stage is in progress, is it possible that we will always see the partial results even after the stage is finished?

Reply:
I used the number of failed tasks as part of key, if failed tasks increase, it will recompute.